ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/util/DataColumnSidecarELRecoveryManagerImplTest.java

```diff
@@ -0,0 +1,233 @@
+
+  final CustodyGroupCountManager custodyGroupCountManager = createCustodyGroupCountManager(4);
+
+  private final DataColumnSidecarELRecoveryManager dataColumnSidecarELRecoveryManager =
+      new PoolFactory(metricsSystem)
+          .createDataColumnSidecarELRecoveryManager(
+              spec,
+              asyncRunner,
+              recentChainData,
+              executionLayer,
+              kzg,
+              dataColumnSidecarPublisher,
+              custodyGroupCountManager);
+
+  private UInt64 currentSlot = historicalTolerance.times(2);
+
+  @BeforeEach
+  public void setup() {
+    when(executionLayer.engineGetBlobs(any(), eq(currentSlot)))
+        .thenReturn(SafeFuture.completedFuture(List.of()));
+    when(kzg.computeCellsAndProofs(any())).thenReturn(kzgCellAndProofs);
+    setSlot(currentSlot);
+  }
+
+  private void setSlot(final UInt64 slot) {
+    currentSlot = slot;
+    dataColumnSidecarELRecoveryManager.onSlot(slot);
+    when(recentChainData.computeTimeAtSlot(any())).thenReturn(UInt64.ZERO);
+  }
+
+  @Test
+  public void onNewBlock_startsFromFuluBlocks() {
+    final Spec minimalWithFuluForkEpoch =
+        TestSpecFactory.createMinimalWithFuluForkEpoch(UInt64.ONE);
+    final DataColumnSidecarELRecoveryManager dataColumnSidecarELRecoveryManagerCustom =
+        new PoolFactory(metricsSystem)
+            .createDataColumnSidecarELRecoveryManager(
+                minimalWithFuluForkEpoch,
+                asyncRunner,
+                recentChainData,
+                executionLayer,
+                kzg,
+                dataColumnSidecarPublisher,
+                custodyGroupCountManager);
+    final SignedBeaconBlock block = dataStructureUtil.randomSignedBeaconBlock(UInt64.ONE);
+    dataColumnSidecarELRecoveryManagerCustom.onSlot(UInt64.ONE);
+    dataColumnSidecarELRecoveryManagerCustom.onNewBlock(block, Optional.empty());
+    assertThat(asyncRunner.hasDelayedActions()).isFalse();
+    verifyNoInteractions(executionLayer);
+
+    final int fuluSlot = minimalWithFuluForkEpoch.slotsPerEpoch(UInt64.ZERO) + 1;
+    final SignedBeaconBlock block2 = dataStructureUtil.randomSignedBeaconBlock(fuluSlot);
+    dataColumnSidecarELRecoveryManagerCustom.onSlot(UInt64.valueOf(fuluSlot));
+    dataColumnSidecarELRecoveryManagerCustom.onNewBlock(block2, Optional.empty());
+    assertThat(asyncRunner.hasDelayedActions()).isTrue();
+    asyncRunner.executeQueuedActions();
+    verify(executionLayer).engineGetBlobs(any(), any());
+  }
+
+  @Test
+  public void onNewBlock_ignoresLocal() {
+    final SignedBeaconBlock block =
+        dataStructureUtil.randomSignedBeaconBlock(currentSlot.longValue());
+    dataColumnSidecarELRecoveryManager.onSlot(currentSlot);
+    dataColumnSidecarELRecoveryManager.onNewBlock(block, Optional.of(RemoteOrigin.LOCAL_PROPOSAL));
+    assertThat(asyncRunner.hasDelayedActions()).isFalse();
+    verifyNoInteractions(executionLayer);
```

Comment:
Again `verifyNoInteractions` probably not necessary.

Reply:
I'd prefer to leave it, see comment above.